enhancements/sig-architecture/33-hub-templates-securing-sensitive-data/README.md

```diff
@@ -0,0 +1,268 @@
+particular AES-CBC with a 256 bit key). The reason is that RSA asymmetric
+encryption is limited to roughly the bit size of the key or less. For instance
+an RSA 4096 key is limited to roughly 4096 bits of encryption. There are
+workarounds such as encrypting several chunks of the data individually, but this
+is much less efficient than just using an algorithm such as AES-CBC. It does add
+some risk since an OpenShift user that does have access to the symmetric key
+stored on the Hub and the replicated policy would be able to decrypt the
+contents stored in the replicated policy even if they didn’t have access to the
+source of the encrypted content. This is easily mitigated with the principle of
+least privilege and is a beneficial tradeoff considering the reduced complexity,
+improved encryption security, and performance gains over asymmetric encryption.
+This is why it is proposed to move forward with using AES-256 encryption.
+
+To avoid identifying the same encrypted data across different policies, a random
+initialization vector (IV) per policy will be used. This will be stored as an
+annotation on the replicated policy since it does not need to be private. This
+implementation does make it so that the duplicate plaintext values that are
+encrypted in a policy for a particular managed cluster would appear the same
+since they would use the same IV and AES key. This added risk is minimal though
+since this can already be determined by looking at the original root policy and
+seeing the Hub policy templates referencing the same key (i.e. field name) of
+the data source (e.g. a Kubernetes secret).
+
+#### Key Generation and Storage
+
+The AES-256 key needs to be generated and stored both on the Hub cluster and the
+managed cluster. There would be a unique key per managed cluster. This would
+mean that if a single managed cluster is compromised, it does not mean that all
+encrypted values in the other managed clusters have also been compromised.
+
+The Policy Propagator should be responsible for generating the AES-256 key for
+each managed cluster. Each key would be stored in a secret called
+`policy-encryption-key` in each respective managed cluster namespace. The key
+would be stored on the secret’s `key` property. This secret should also have the
+`cluster.open-cluster-management.io/backup: 'policy'` label set so that the key
+is backed up during backup and restore situations. With this in place, the
+Policy Propagator is able to encrypt the data using the `protect` template
+function, however, the managed cluster also needs the key when it decrypts the
+data. The Policy Spec Sync controller should also sync the
+`policy-encryption-key` secret to the managed cluster in the managed cluster
+namespace by watching the managed cluster namespace for secrets on the Hub. It
+is also desired to allow customers to provide their own keys. This would be
+limited to AES-256 keys unless a customer requests otherwise.
```

Comment:
If this key is modified or deleted, does it mean the policy on spoke can not be decrypted? Will it transport key once or multiple times?

Reply:
The secret is constantly synced from the Hub to the managed cluster. I'll add that detail in the doc.